cmd/ui/src/rendering/utils/utils.ts

```diff
@@ -43,3 +45,78 @@ export const getNodeRadius = (highlighted: boolean, inverseSqrtZoomRatio: number
 
     return radius;
 };
+
+//This was sourced from https://pomax.github.io/bezierinfo/#circleintersection where the steps are explained in detail
+export const getCurveCircleIntersection = (
+    LUT: Coordinates[],
+    circleCenter: Coordinates,
+    radius: number,
+    deltaEpsilon = 0.01
+) => {
+    const values = [];
+    const maxIterations = 25;
+    let start = 0,
+        count = 0;
+
+    while (++count < maxIterations) {
+        if (!LUT[start - 2] || !LUT[start - 1]) {
+            start = start + 1;
+            continue;
+        }
+        const closest = findClosest(
+            circleCenter.x,
+            circleCenter.y,
+            LUT.slice(start),
+            bezier.getLineLength(LUT[start - 2], circleCenter),
+            bezier.getLineLength(LUT[start - 1], circleCenter),
+            radius,
+            deltaEpsilon
+        );
+        const i = start + closest;
+
+        if (i < start) break;
+        if (i > 0 && i === start) break;
+        values.push(i);
+        start = i + 2; // We need to add 2, because findClosest already tells us that i+1 cannot be a viable candidate.
+    }
+    return values;
+};
+
+const findClosest = (
+    x: number,
+    y: number,
+    LUT: Coordinates[],
+    pd2: number,
+    pd1: number,
+    radius: number,
+    distanceEpsilon: number
+): number => {
+    let distance = Infinity,
+        prevDistance2 = pd2 || distance,
+        prevDistance1 = pd1 || distance,
+        i = -1;
+
+    for (let index = 0; index < LUT.length; index++) {
+        const pointOnCurve = LUT[index];
+        const pointDistance = Math.abs(
+            bezier.getLineLength({ x: x, y: y }, { x: pointOnCurve.x, y: pointOnCurve.y }) - radius
+        );
+
+        // Realistically, there's only going to be an intersection if
+        // the distance to the circle center is already approximately
+        // the circle's radius.
+        if (prevDistance1 < distanceEpsilon && prevDistance2 > prevDistance1 && prevDistance1 < pointDistance) {
```

Comment:
If nodes are past a certain distance from each other, the arrows are no longer drawn and the `curve circle intersection not found for drawing arrowhead` warning gets thrown in the console. It seems to be because at a certain distance, these conditions listed here are no longer being met and `findClosest` always returns `-1`.

My half-baked theory is that this is because our lookup tables are based on a constant resolution, and at a certain point there are no values in the lookup table close enough to the circles radius, but I am not 100% sure.

Reply:
Yea I think you're right. Upping the resolution should work but needs more computation. I may be able to mitigate some of that by only evaluating in narrowed windows for `t` since we know that the nodes should be close to 0 or 1 values of `t` and will never be at 0.5 `t` for instance. I'll try this out